### PR TITLE
feat: Adds new ErrorResponse class

### LINKED ISSUE #798: Create a custom ErrorResponse class
This custom error class will be responsible for providing details on errors sent to the front-end. The JSON format of the ErrorResponse class is below.

```JSON
{
    error_identifier: "login.incorrectPassword", 
    default_message: "Your password is incorrect."
}
```

`default_message` should be optional. For v3 of the API the front-end should always expect an HTTP status code and an optional ErrorResponse. We plan to slowly roll out ErrorResponse to more errors over time. In regards to i18n, the front-end should handle localization whenever possible, if not possible the back-end may offer a localized default message but this isn't always guaranteed.

### PR DESCRIPTION
**Pull request by cbaudouinjr**
Adds a new ErrorResponse class for better communicating errors to our front-end. This class includes an error_identifier and default_message. To create, all you need to do is call `ErrorResponse.new` and supply an error symbol which are pulled from the `error_resposne.rb` class. HTTP status codes should continue to be sent as they normally were. 

The ErrorResponse creation line and HTTP status code can both be written as a one-liner for simplicity:
`render json: ErrorResponse.new(:login_merge_newSchoolNameMissing), status: :bad_request`

This pull request shows one usage example with the `school_controller.rb` controller, specifically the merge schools functionality which may fail for several different reasons.

Closes #798

### CODE REVIEW

File: lib/v3/error_response.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class ErrorResponse
+  attr_accessor :error_identifier, :default_message
+
+  ERRORS = [:login_merge_newSchoolNameMissing].freeze
```
Comment (peterkos) [inline lib/v3/error_response.rb:4]: 🙏

Comment (peterkos) [reply lib/v3/error_response.rb:4]: How are we going to document these?

File: lib/v3/error_response.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+  def initialize(error_identifier, default_message = nil)
```
Comment (peterkos) [inline lib/v3/error_response.rb:6]: Maybe specify in docs that `default_message` is sent back in the error response JSON? Or that might just be obvious

File: app/controllers/manage/schools_controller.rb [hunk 3]
```diff
@@ -38,8 +38,7 @@ def destroy
   def perform_merge
     new_school_name = params[:school][:id]
     if new_school_name.blank?
-      # FIXME: This might need to be :unprocessable_entity
-      head :bad_request
+      render json: ErrorResponse.new(:login_merge_newSchoolNameMissing), status: :bad_request
```
Comment (peterkos) [inline app/controllers/manage/schools_controller.rb:41]: I really love this; super clean and readable :smile:

**Review by peterkos — APPROVED** [review]
LGTM!

File: lib/v3/error_response.rb (hunk 1 above)
Comment (cbaudouinjr) [reply lib/v3/error_response.rb:4]: #780 should handle this, if not auto generated we'll add manually.

**Pull request merged**